website_form_recaptcha/models/website_form_recaptcha.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2015-2017 LasLabs Inc.
+# Copyright 2019 Simone Orsi - Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, http, models
+from odoo.exceptions import ValidationError
+import logging
+import requests
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteFormRecaptcha(models.AbstractModel):
+    """ This model provides ReCaptcha helper methods.
+    Nothing is stored in the DB.
+    """
+
+    _name = 'website.form.recaptcha'
+    _description = 'Website Form Recaptcha Validations'
+    URL = 'https://www.google.com/recaptcha/api/siteverify'
+    RESPONSE_ATTR = 'g-recaptcha-response'
+    # name of the token attr to store on the request object
+    REQUEST_TOKEN = RESPONSE_ATTR.replace('-', '_')
+
+    @api.model
+    def _get_error_message(self, errorcode=None):
+        mapping = {
+            'missing-input-secret': _('The secret parameter is missing.'),
+            'invalid-input-secret':
+                _('The secret parameter is invalid or malformed.'),
+            'missing-input-response': _('The response parameter is missing.'),
+            'invalid-input-response':
+                _('The response parameter is invalid or malformed.'),
+        }
+        return mapping.get(errorcode, _('There was a problem with '
+                                        'the captcha entry.'))
+
+    def _get_api_params(self):
+        ICP = self.env['ir.config_parameter'].sudo()
+        return {
+            'site_key': ICP.get_param('recaptcha.key.site'),
+            'secret_key': ICP.get_param('recaptcha.key.secret'),
```

Comment:
These should come from the `website` object, right?

Reply:
no, as explained above they can come via params